pygyro/initialisation/initialiser_funcs.py

```diff
@@ -1,75 +1,106 @@
-from pyccel.decorators import types, pure
-
-@pure
```

Comment:
Why did you remove the `@pure` decorator? This is useful for the fortran compiler to make optimisations. See https://github.com/pyccel/pyccel/blob/master/tutorial/decorators.md#pure

Reply:
Yes, I will readd them